crates/astria-core/src/sequencerblock/v1alpha1/block.rs

```diff
@@ -1314,13 +1321,17 @@ impl Deposit {
         amount: u128,
         asset: asset::Denom,
         destination_chain_address: String,
+        id_of_source_transaction: TransactionId,
```

Comment:
Comment on the containing `Deposit::new`: this constructor is becoming unwieldy. I am fine if you keep it as is for this PR, but can you remove `Deposit::new` in favor of `Deposit::try_from_raw` (which is mostly done in tests anyway)?

Reply:
Can you make an issue for this to fix in a followup?

Reply:
Do you mean to remove `Deposit::new` entirely? Or just change the instances which are in tests to use `Deposit::try_from_raw`? The `new()` method is used in some actions for creating deposits, so it seems to me like it may be of some use to keep it for these instances, and if we already have it then IMO we may just as well use it for tests. Apologies if I'm misunderstanding what you're asking

Reply:
The problem is with the method itself: a constructor with 6+ arguments is just terrible and is usually replaced by a builder.